Source/driver/Castro.cpp

```diff
@@ -3406,6 +3406,52 @@ Castro::apply_tagging_restrictions(TagBoxArray& tags, Real time)
 
     }
 #endif
+
+    auto geomdata = geom.data();
+
+    // Allow the user to limit tagging outside of some distance from the problem center.
+#ifdef _OPENMP
+#pragma omp parallel
+#endif
+    for (MFIter mfi(tags); mfi.isValid(); ++mfi) {
+        const Box& bx = mfi.tilebox();
+
+        auto tag = tags[mfi].array();
+
+        amrex::ParallelFor(bx,
+        [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k)
+        {
+            const Real* problo = geomdata.ProbLo();
+            const Real* probhi = geomdata.ProbHi();
+            const Real* dx = geomdata.CellSize();
+
+            Real loc[3] = {0.0};
+
+            loc[0] = problo[0] + (static_cast<Real>(i) + 0.5_rt) * dx[0];
+#if AMREX_SPACEDIM >= 2
+            loc[1] = problo[1] + (static_cast<Real>(j) + 0.5_rt) * dx[1];
+#endif
+#if AMREX_SPACEDIM == 3
+            loc[2] = problo[2] + (static_cast<Real>(k) + 0.5_rt) * dx[2];
+#endif
+
+            Real r = std::sqrt((loc[0] - problem::center[0]) * (loc[0] - problem::center[0]) +
+                               (loc[1] - problem::center[1]) * (loc[1] - problem::center[1]) +
```

Comment:
is center really defined for 3 dimensions even if AMREX_SPACEDIM < 3?

Reply:
Yes -- because this variable is defined now from _prob_params, so it is sized to the maximum it could be.